### ISSUE TITLE
Allow Traditional WebJobs to run as Azure Functions

### ISSUE THREAD

**Issue by JSkimming** [Azure/azure-webjobs-sdk-script#1379]
Will you be providing the ability to run traditional WebJobs as Azure Functions, for instance Queue/Blob/EventHub triggered events?

Or, as what may be more likely given a [recent blog post](https://blogs.msdn.microsoft.com/appserviceteam/2017/03/16/publishing-a-net-class-library-as-a-function-app/ "Publishing a .NET class library as a Function App"), will I be able to publish a class library that allows all the same extensibility as through the WebJobs SDK.

Where I'm coming from is this. I like the flexibility of Web Jobs (e.g. to extend the logging to a [different logging framework](https://github.com/JSkimming/serilog-extensions-webjobs "Serilog.Extensions.WebJobs")), but I also like the Consumption Plan pricing model of Functions.

I appreciate there will have to be some limitations, for instance continuously running jobs would not work with the Consumption Plan, but some of the extensibility points may still be a possibility.

**Comment by lindydonna** [comment]
@JSkimming We're moving in this direction with some feature improvements we're doing for the Functions Visual Studio tooling. There are some features that aren't yet supported in Functions (e.g., the Singleton attribute), but most use cases will apply.

**Comment by JSkimming** [comment]
@lindydonna thanks for reply, I wasn't sure whether this was the correct place to put my question.

I'm interested in keeping up to date with the current features and future roadmap of functions and WebJobs. Is the [Azure App Service Team Blog](https://blogs.msdn.microsoft.com/appserviceteam/ "How the App Service team functions") the best place to look?

**Comment by lindydonna** [comment]
@JSkimming Yes, and you can also follow the Twitter account [@AzureFunctions](https://twitter.com/azurefunctions).

**Comment by MikeStall** [comment]
"will I be able to publish a class library that allows all the same extensibility as through the WebJobs SDK."
@JSkimming  - yes, with some minor caveats.

**Comment by JSkimming** [comment]
@MikeStall I've been using the [VS tooling](https://github.com/Azure/azure-webjobs-sdk-script/issues/1379#issuecomment-293063522) as descrbed by @lindydonna to develop an Azure Functions class library.

I've been digging into how the WebJobs extensions are available and whether I could use [my own extension](https://github.com/JSkimming/serilog-extensions-webjobs) in Azure Functions.

From what I can tell, the Azure Functions runtime is [hard coded to add _known_ extensions](https://github.com/Azure/azure-webjobs-sdk-script/blob/v1.0.11002/src/WebJobs.Script/Host/ScriptHost.cs#L729-L734). Is there a current or planned extensibility point where I can register a third party extension. Or Are you amenable to to pull requests to add extensions to the _"known extension assemblies"_?

**Comment by MikeStall** [comment]
@JSkimming  -  we have a preview of a new mechanism for dynamically loading new user extensions. Currently, it's just an appsetting pointing to some dlls. See https://github.com/Azure/WebJobsExtensionSamples for sample and docs. 
Proceed with caution - it's preview and we still have a a few assembly loading bugs to work out (https://github.com/Azure/azure-webjobs-sdk-script/labels/feature%3Abyob )

**Comment by JSkimming** [comment]
@MikeStall - Thanks for the pointers, I'll be taking a look.

Regarding assembly loading bugs, I've previously implemented [something similar](https://github.com/AutoTestNET/AutoTest.ArgumentNullException/blob/v0.8.0/src/AutoTest.ArgumentNullException/Framework/ReflectionDiscoverableCollection.cs "ReflectionDiscoverableCollection"), which I liflted from the [Glimpse project](https://github.com/Glimpse/Glimpse/blob/1.8.6/source/Glimpse.Core/Framework/ReflectionDiscoverableCollection.cs "ReflectionDiscoverableCollection").

There's also other gotchas, for which @haacked [previously blogged about](http://haacked.com/archive/2012/07/23/get-all-types-in-an-assembly.aspx/ "Get All Types in an Assembly").

Feel free to ignore, just thought I'd share 😃
